### PR TITLE
Add docs badges to the concepts pages

### PR DESCRIPTION
**Pull request by rivertam**
Again, Prettier did stuff automatically. I think it's for the best, but
I can still take it out if you'd prefer.

I checked every link I added, but a double check would not be
unwelcome.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/849)

### CODE REVIEW

**Comment by Moxinilian** [comment]
Can we see a screenshot of what it does?

**Comment by rivertam** [comment]
@Moxinilian Screenshot added to the PR comment

**Comment by Moxinilian** [comment]
What do theses badges do exactly?

**Comment by jojolepro** [comment]
Links to develop and master docs of shrev

**Comment by Moxinilian** [comment]
I'm not sure it is a good idea to be able to swap between develop and master so easily, it could create confusion.

**Comment by rivertam** [comment]
They just link to the respective docs of each concept. I often get frustrated reading tutorials that don't link to the API documentation for more details.

**Comment by jojolepro** [comment]
You should link to the exact version used by this amethyst release.

**Comment by rivertam** [comment]
I'm actually not even seeing a way to visit old versions of the docs. Do you know how I can do that? I tried going to like `https://www.amethyst.rs/doc/0.6.0/doc/amethyst/` etc. to no avail.

**Comment by jojolepro** [comment]
You have to go on docs.rs for that. On the amethyst.rs website we only host the master and develop branches.

**Comment by LucioFranco** [comment]
@rivertam I like this idea, but I do not like having both master and develop on the same page. I don't know if there is a way a variable can be passed down in mdbook but if that is possible, I'd like to have the book on develop point just to the develop docs.

So one solution is to either embed the docs within the book so that it can be reached by a relative link. But I don't think having both is the right idea, as @jojolepro it will be just confusing.

**Comment by rivertam** [comment]
@LucioFranco I've raised the issue on the mdBook repository for configurable variables. https://github.com/rust-lang-nursery/mdBook/issues/760

I could also write a specific preprocessor for this project, but that would require amethyst to move to a custom mdBook implementation rather than the CLI.

**Comment by LucioFranco** [comment]
@rivertam Great, I'm curious to see how the preprocessor might be implemented. I'm not really a big fan of having a custom implementation that we will have to maintain. If we can get mdBook to insert a variable either from the environment or via a setting in the `book.toml` I would prefer that as travis can do all the work there.

Now, something we could do now is update amethyst/website to somehow insert the correct version using something like `sed` into the generated html.

**Comment by rivertam** [comment]
@LucioFranco That's a good point -- we could definitely do this in JavaScript.

**Comment by LucioFranco** [comment]
@rivertam that's not a bad idea, if it can be done without compromising anything, I'm all for it!

File: book/src/concepts/dispatcher.md [hunk 1]
```diff
@@ -14,5 +14,4 @@ The dispatcher looks at all the `SystemData` in the systems and builds execution
 
 If you want to have the best performance possible, you should prefer immutable over mutable whenever it is possible. (`Read` instead of `Write`, `ReadStorage` instead of `WriteStorage`).
 
-_Note: Please however keep in mind that `Write` is still preferable to interior mutability, such as `Mutex` or `RwLock` for example.
-
+\_Note: Please however keep in mind that `Write` is still preferable to interior mutability, such as `Mutex` or `RwLock` for example.
```
Comment (torkleyy) [inline book/src/concepts/dispatcher.md]: Unrelated to this PR, but this line doesn't make sense. `Write` _is_ interior mutability. Maybe this should be "preferable locks"? And then, there are cases where locks actually are preferable (we do use them for a few things).

Comment (LucioFranco) [reply book/src/concepts/dispatcher.md]: @torkleyy I agree, should we open an issue for that? Since this PR is unrelated?

Comment (jojolepro) [reply book/src/concepts/dispatcher.md]: moved to https://github.com/amethyst/amethyst/issues/870

**Comment by jojolepro** [comment]
If you think support for getting the current library version is not coming soon, what you can do is hardcode the current dependencies versions in for now, and open a PR later with the dynamic versionning. That way we can merge the documentation updates before starting to merge the features PR (this way, we don't have to wait until the next version to fix the docs)

**Comment by jojolepro** [comment]
Hi. Is this still a thing?

**Comment by rivertam** [comment]
Hey @jojolepro It's vaguely still a thing but I'm not really that actively pursuing it right now. I'd rather not hardcode the dependencies versions just because that's a lot of work for myself and a lot of work for you guys when you update the version. So for my needs it's kind of pending https://github.com/rust-lang-nursery/mdBook/issues/760, which I'm also not that actively pursuing.

**Comment by LucioFranco** [comment]
@rivertam cool, we can close this then and if you want to attempt it again you can reopen this or open a new one! Thanks.

**Pull request closed without merging**